lib/salus/scanners/cargo_audit.rb

```diff
@@ -0,0 +1,96 @@
+require 'json'
+# Rust Cargo Audit scanner integration. Flags known malicious or vulnerable
+# dependencies in rust projects that are packaged with cargo.
+# https://github.com/RustSec/cargo-audit
+
+module Salus::Scanners
+  class CargoAudit < Base
+    def should_run?
+      @repository.cargo_lock_present?
+    end
+
+    def run
+      Dir.chdir(@repository.path_to_repo) do
+        shell_return = run_shell(command)
+
+        # Cargo Audit has the following behavior:
+        #
+        # - vulnerability NOT found:
+        #   - status: 0
+        #   - stderr: ""
+        #   - stdout: Milestones, counts of crates scanned
+        #
+        # - vulnerability found:
+        #   - status: 1
+        #   - stderr: ""
+        #   - stdout: JSON detail of the vulnerability
+        #
+        # - Error running audit:
+        #   - status: 1
+        #   - stderr: String with details on the error prenting the run (not JSON)
+        #   - stdout: ""
+
+        return report_success if shell_return.success?
+
+        if shell_return.stderr.empty?
+          report_failure
```

Comment:
Can you move `report_failure` to right before `if shell_return.stderr.empty?`
Like gosec https://github.com/coinbase/salus/blob/master/lib/salus/scanners/gosec.rb#L38

Otherwise, the `else` branch will not be flagged as a failure.

Reply:
Ah good catch - we call report_failure for both vulnerabilities and failed scans.  Added tests cases for the expanded behavior.